### DISCUSSION TITLE
How to retry a build that failed?

### DISCUSSION THREAD

**Discussion by DanTup** [community/community#26037] [Actions]
<p>I had a build fail for a reason unrelated to my code (in the setup-node action - possible a network issue or something, it’s not clear). I can’t rind any way to retry this build without pushing another commit (which means all the other parallel tasks like running on other OSes will be repeated).</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8ae6ac0942ca56a01ad2da3bfc34abd69dd72a0d.png" data-download-href="/uploads/short-url/jOM6pC5aXNIhtnIAJMZOkYX9P1r.png?dl=1" title="Screenshot 2019-09-19 at 12.30.17 pm.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/1078012/181094900-b92942c8-829c-4885-8aea-2ece13b4065e.png" alt="Screenshot 2019-09-19 at 12.30.17 pm.png" data-base62-sha1="jOM6pC5aXNIhtnIAJMZOkYX9P1r" width="690" height="226" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8ae6ac0942ca56a01ad2da3bfc34abd69dd72a0d_2_690x226.png, https://aws1.discourse-cdn.com/github/original/2X/8/8ae6ac0942ca56a01ad2da3bfc34abd69dd72a0d.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/8ae6ac0942ca56a01ad2da3bfc34abd69dd72a0d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8ae6ac0942ca56a01ad2da3bfc34abd69dd72a0d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2019-09-19 at 12.30.17 pm.png</span><span class="informations">938×308 35.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mscoutermarsh** [comment]
<p>Hello!</p>
<p>We currently don’t have the retry button enabled for Actions. 100% agree this is super important. We’re working on it now, won’t be much longer. :grinning:</p>

**Comment by mathstuf** [comment]
<p>This does seem to be there now. However, I’m seeing that I get a “check rerequested” event is in the <code>GITHUB_EVENT_PATH</code> file rather than a <code>synchronize</code> event (or whatever did trigger the action run) causing some assumptions to be broken. It certainly doesn’t have the same information available as the original <code>synchronize</code> trigger at least. It also seems to be only for all actions, not per action (though I’m just testing with a single action right now).</p>

**Comment by mscoutermarsh** [comment]
<p>:+1: thanks! We’re working on it.</p>

**Comment by runspired** [comment]
<p>Thanks! Retry is hugely important to us :slight_smile: The ability to retry individual tasks/checks as well :) </p>

**Comment by chrispat** [comment]
<p>For future runs this should now be solved <a href="https://github.blog/changelog/2019-09-30-github-actions-deterministic-re-runs-for-workflows/" rel="nofollow noopener">https://github.blog/changelog/2019-09-30-github-actions-deterministic-re-runs-for-workflows/</a></p>

**Comment by DanTup** [comment]
<p>I don’t understand how re-runs work. When I have a failed task, there’s an info icon next to it with a tooltip that says “This check suite has disabled re-running individual check runs”. Is there something I need to do to enable this?</p>

**Comment by FacetGraph** [comment]
<p>Will we be able to retry specific jobs rather than the entire workflow? Right now I only see the option to Re-run all checks. If I am running multiple jobs in parallel and only one fails I would like the ability to rerun just that one job.</p>

**Comment by purefan** [comment]
<p>I cant find it and it’s now 6 months since your message, am I missing it or is Github still working on it?</p>

**Comment by mangelajo** [comment]
<p>We don’t see a re-run command sometimes. <a href="https://github.com/submariner-io/shipyard/runs/666918597?check_suite_focus=true" rel="nofollow noopener">https://github.com/submariner-io/shipyard/runs/666918597?check_suite_focus=true</a></p>
<p>I remember seeing it in the past. It would be nice to be able to re-run specific tests.</p>

**Comment by tarc** [comment]
<p>Although the possibility to rerun failed workflows is great, it would be super great if there’s an option to run only the failed jobs. Is this already there and I’m missing it? Is there a plan to implement it?</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Updating this, as my old comment is very out of date at this point.</p>
<p>Re-running an entire workflow is possible via the re-run button on the top right or the <a href="https://developer.github.com/v3/actions/workflow-runs/#re-run-a-workflow" rel="nofollow noopener">Actions API</a>. The re-run button will trigger the entire workflow to run again.</p>
<p>In the future, we’ll be making this more specific and allowing for individual jobs within a workflow to be re-run.</p>

**Comment by johan-bjareholt** [comment]
<aside class="quote group-github-employees" data-username="mscoutermarsh" data-post="12" data-topic="16297">
<div class="title">
<div class="quote-controls"></div>
 mscoutermarsh:</div>
<blockquote>
<p>Re-running an entire workflow is possible via the re-run button on the top right</p>
</blockquote>
</aside>
<p>If you like me think that “Hey this is not true, there is no re-run button!” it’s likely that your job is not finished yet and that “re-run” button is actually a “cancel” button. So you just need to cancel your job before you can re-run it.</p>

**Comment by loopy3025** [comment]
We have checks that run via webhooks. The Actions API is not enabled, so we don't have a retry button. We can re-run webhooks, but the webhooks interface only shows the 17 most "Recent Deliveries", so I can't actually re-deliver anything older.

The use case here is when code from one PR fixes a build issue with many other PRs.  When the fixing PR is merged, the rest of the PRs need to be rebased with the new merged base branch. It's not always feasible to push an empty commit to every PR because they are usually all branches sitting on various developers' forks.

**Reply by loopy3025** [reply]
Honestly, the great thing would be for PRs to rebase to the base branch's new HEAD automagically whenever a pull request is merged.

**Reply by mathstuf** [reply]
Something like that needs human interaction as otherwise conflict messes are likely to accumulate. A request for a robot to do it sounds fine, but automagic means that if you're landing a series of PRs, you have to wait for pipelines to refresh after each merge (wasting a lot of CI resources as even further PRs are going to spin CI for each PR that lands ahead of it).
